rasa/nlu/model.py

```diff
@@ -77,17 +77,16 @@ def load(model_dir: Text):
         try:
             metadata_file = os.path.join(model_dir, "metadata.json")
             data = rasa.shared.utils.io.read_json_file(metadata_file)
-            return Metadata(data, model_dir)
+            return Metadata(data)
         except Exception as e:
             abspath = os.path.abspath(os.path.join(model_dir, "metadata.json"))
             raise InvalidModelError(
                 f"Failed to load model metadata from '{abspath}'. {e}"
             )
 
-    def __init__(self, metadata: Dict[Text, Any], model_dir: Optional[Text]):
+    def __init__(self, metadata: Dict[Text, Any]):
```

Comment:
```suggestion
    def __init__(self, metadata: Dict[Text, Any]) -> None:
```

Reply:
It's not really needed. It's only needed if your init method doesn't have any arguments and you still want it to be type checked, as far as I recall. And also, `-> NoReturn` would be more accurate anyway

Reply:
https://www.python.org/dev/peps/pep-0484/ says

> Note that the return type of __init__ ought to be annotated with -> None

I think it would be more consistent to annotate it everywhere with `None`.

Reply:
Directly quoting the boss -> https://github.com/python/mypy/issues/3358#issuecomment-465783430 (doesn't happen every day ahah)